lib/Sema/CSSimplify.cpp

```diff
@@ -2063,13 +2063,17 @@ static bool isInPatternMatchingContext(ConstraintLocatorBuilder locator) {
     if (path.back().is<LocatorPathElt::PatternMatch>()) {
       return true;
     } else if (path.size() > 1) {
+      auto pathElement = llvm::find_if(path, [](LocatorPathElt &elt) {
+          return elt.is<LocatorPathElt::PatternMatch>();
+      });
+      
+      if (pathElement == path.end())
+          return false;
       // sub-pattern matching as part of the enum element matching
       // where sub-element is a tuple pattern e.g.
       // `case .foo((a: 42, _)) = question`
-      auto lastIdx = path.size() - 1;
-      if (path[lastIdx - 1].is<LocatorPathElt::PatternMatch>() &&
-          path[lastIdx].is<LocatorPathElt::FunctionArgument>())
-        return true;
+      if (pathElement->is<LocatorPathElt::PatternMatch>())
```

Comment:
This is redundant because we know if we found an element this is always true at this point.